File: tests/TimedTest/cpp/Filter.cc
#include <Filter.h>

namespace timedtest {
    namespace components {

        Filter::Filter(_component_conf &config, riaps::Actor &actor) :
                FilterBase(config, actor) {
        }

        void Filter::OnReady(const DummyT::Reader &message,
                             riaps::ports::PortBase *port) {
            if (port->GetPortBaseConfig()->isTimed) {
                _logger->info("{}, sendTs: {}.{}, recvTs: {}.{}", __FUNCTION__,
                                 port->AsRecvPort()->GetLastSendTimestamp().tv_sec,
                                 port->AsRecvPort()->GetLastSendTimestamp().tv_nsec,
                                 port->AsRecvPort()->GetLastRecvTimestamp().tv_sec,
                                 port->AsRecvPort()->GetLastRecvTimestamp().tv_nsec);
            } else {
                _logger->info("{}", __FUNCTION__);
            }
            _logger->info("{}", message.getMsg().cStr());

            capnp::MallocMessageBuilder builder;
            auto msg = builder.initRoot<DummyT>();
            msg.setMsg("FromFilter");
            if (SendQuery(builder, msg)){
                DummyT::Reader reader;
                RecvQuery(reader);
            }
        }


        void Filter::OnGroupMessage(const riaps::groups::GroupId &groupId,
                                    capnp::FlatArrayMessageReader &capnpreader, riaps::ports::PortBase *port) {

        }

        Filter::~Filter() {

        }
    }
}

riaps::ComponentBase *create_component(_component_conf &config, riaps::Actor &actor) {
    auto result = new timedtest::components::Filter(config, actor);
    return result;
}

void destroy_component(riaps::ComponentBase *comp) {
    delete comp;
}

File: tests/TimedTest/python/Filter.py
#Filter.py
from riaps.run.comp import Component
import os
import logging

class Filter(Component):
    def __init__(self):
        super(Filter, self).__init__()	        
        self.pid = os.getpid()
        self.logger.info("(PID %s) - starting Filter",str(self.pid))
        

    def on_ready(self):
        msg = self.ready.recv_pyobj()
        self.logger.info("PID (%s) - on_ready():%s",str(self.pid),str(msg))

    def on_query(self):
        req = self.query.recv_pyobj()
        self.logger.info("PID (%s) - on_query():%s",str(self.pid),str(req))
    
    def __destroy__(self):			
        self.logger.info("(PID %s) - stopping Filter",str(self.pid))   	        	        
